python/lsst/ts/salobj/domain.py

```diff
@@ -274,45 +273,44 @@ def remove_salinfo(self, salinfo):
         except KeyError:
             return False
 
+    def basic_close(self):
+        """A synchronous and less thorough version of `close`.
+
+        Intended for exit handlers and constructor error handlers.
+        """
+        try:
+            while self._salinfo_set:
+                salinfo = self._salinfo_set.pop()
+                salinfo.basic_close()
+        finally:
+            self.participant.close()
+        self.participant.close()
+
     async def close(self):
         """Close all registered `SalInfo` and the dds domain participant.
 
         May be called multiple times. The first call closes the Domain;
         subsequent calls wait until the Domain is closed.
         """
-        if self.participant is None:
+        if not self.isopen:
             await self.done_task
             return
+        self.isopen = False
         try:
             while self._salinfo_set:
                 salinfo = self._salinfo_set.pop()
                 await salinfo.close()
         finally:
-            self.close_dds()
+            self.participant.close()
```

Comment:
This bit of code from line 298 to here is exactly like the one in `basic_close`, consider calling it instead (if what you want to do is a "basic_close" here.

Reply:
Good suggestion.

Reply:
It is not quite the same: this `Domain.close` code calls `await salinfo.close`, whereas `Domain.basic_close` calls `salinfo.basic_close()` synchronously. `SalInfo.close` does a bit more.